.github/workflows/main.yml

```diff
@@ -10,6 +10,26 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_changed_folders:
+    name: Check files
+    outputs:
+      run_job: ${{ steps.check_files.outputs.run_job }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: check modified files
+        id: check_files
+        run: |
+          files=`git diff --name-only HEAD~1 | awk -F "/*[^/]*/*$" '{ print ($1 == "" ? "." : $1); }' | sort | uniq`
```

Comment:
Here is probably not the best place to implement. I'd go for a plugin to an existing linter that would be called conditionally. And avoid shell-scripting in favor of Python. Also, this won't catch changes happening more than one commit away. People often add change note as a separate commit post PR creation and this would only be linting that commit that only adds a change note.